### ISSUE TITLE
Slack official Plugin: Job exception: uninitialized constant Jobs::NotifySlack

### ISSUE THREAD

**Issue by Nighthawk22** [bitnami/bitnami-docker-discourse#29]
**Description**

Checkout of the official-slack-plugin into the plugins folder. Aftwards i restarted the container and the plugin appeared in the admin panel.
The problem is that the sidekiq is not processing the slack notifcations and produces the following error:
```
Job exception: uninitialized constant Jobs::NotifySlack
```

How can i fix this?
**Steps to reproduce the issue:**

1. Bring up discourse
2. Checkout the https://github.com/discourse/discourse-slack-official plugin
3. Setup the plugin
4. Create a new topic and wait for the slack notification
5. Look into the /opt/bitnami/discourse-sidekiq/logs/production.log at the sidqkiq container

**Describe the results you received:**

```
Job exception: uninitialized constant Jobs::NotifySlack
```
Error and sidekiq is not processing anymore

**Describe the results you expected:**

Slack notification is send and sidkiq is working the queues

**Additional information you deem important (e.g. issue happens only occasionally):**
I am running on an kubernetes cluster

**Version**

- Output of `docker version`:

```
Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:19:04 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

- Output of `docker info`:

```
Containers: 52
 Running: 39
 Paused: 0
 Stopped: 13
Images: 82
Server Version: 17.06.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 563
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.10.0-30-generic
Operating System: Ubuntu 16.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.795GiB
Name: taa-kube-node-1
ID: LG6V:TSC3:LR2B:GISB:LPBQ:JCRC:UTJO:O6RW:5IYL:WUUQ:HBWY:HLPA
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.5", GitCommit:"17d7182a7ccbb167074be7a87f0a68bd00d58d97", GitTreeState:"clean", BuildDate:"2017-08-31T19:32:26Z", GoVersion:"go1.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.3", GitCommit:"2c2fe6e8278a5db2d15a013987b53968c743f2a1", GitTreeState:"clean", BuildDate:"2017-08-03T06:43:48Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
Self hosted kubernetes cluster version 1.7.3

**Comment by chipironcin** [comment]
Hi @Nighthawk22 
Sorry for the late response. We are catching up with the raised issues in our images.

I was able to reproduce the issue.
As a workaround, just perform the same steps in the `discourse-sidekiq` container (aka. clone the slack plugin repo into the plugins directory).

We will work towards sharing a volume between the two containers so this works out of the box. Will keep this thread updated ;)

**Comment by Nighthawk22** [comment]
Hi!
Thanks for the response. That was exactly what I did afterwards.

I additionally experienced that the sidekiq does not process all queues. I needed to extend the sidekiq.yml and add:
```
production:
   queues:
      - [critical, 4]
      - [default, 2]
      - [low]
```
Otherwise critical and low are not processed.

**Comment by chipironcin** [comment]
Noted, thanks for your feedback!

**Comment by AndreaGiardini** [comment]
@chipironcin I had troubles debugging this problem for several hours. Can this be fixed upstream? the sidekiq.yml file does not process queues of the production environment.

**Comment by dbarranco** [comment]
Hi @AndreaGiardini 

@chipironcin  created an internal ticket to check this and have this configuration working out of the box.
As soon as we have any news on this we will update this post.

Sorry for the inconveniences it may cause.

Kind regards.

**Comment by BirknerAlex** [comment]
Hello,

this issues still exists. A fix in the upstream would be great.

**Comment by beltran-rubo** [comment]
This issue has been solved in the version 2.0.3-r7. @BirknerAlex @AndreaGiardini @Nighthawk22  See this for more details:

https://github.com/bitnami/bitnami-docker-discourse/issues/40#issuecomment-410186599
